docs/src/main/asciidoc/kafka-streams.adoc

```diff
@@ -722,3 +722,226 @@ For more details about the health information, see the
+                        .toStream()));
+    }
+----
+
+In the above function bean, there are 2 inputs and one input. The function that returns from the method takes a `KStream` as input, but if you look at the output that is another function
+which takes a `KTable` as its input. The output of this second function is a `KStream` which becomes the output of the procesor.
+Both inputs are available as references in the method body and the applications can perform various operations on them.
+In this example we use function currying on two partial functions.
+One thing to keep in mind is that the input bindings must follow a natural order of sorting when you have multiple input bindings, otherwise the binder won't know which binding to bind for the various function inputs.
+Here is the corresponding binding interface for the above processor.
+
+[source]
+----
+interface KStreamKTableProcessor {
+
+    @Input("input-1")
+    KStream<?, ?> input1();
+
+    @Input("input-2")
+    KTable<?, ?> input2();
+
+    @Output("output")
+    KStream<?, ?> output();
+
+}
+----
+
+If you look at the 2 inputs, there is a natural sorting order - i.e. input-1 goes to the first partial function input and input-2 goes to the second partial function.
+
+Here is another example that shows multiple inputs with GlobalKTable.
+
+[source]
+----
+	@Bean
+    public Function<KStream<Long, Order>,
+            Function<GlobalKTable<Long, Customer>,
+                    Function<GlobalKTable<Long, Product>, KStream<Long, EnrichedOrder>>>> process() {
+
+        return orderStream -> (
+                customers -> (
+                        products -> (
+                                orderStream.join(customers,
+                                        (orderId, order) -> order.getCustomerId(),
+                                        (order, customer) -> new CustomerOrder(customer, order))
+                                        .join(products,
+                                                (orderId, customerOrder) -> customerOrder
+                                                        .productId(),
+                                                (customerOrder, product) -> {
+                                                    EnrichedOrder enrichedOrder = new EnrichedOrder();
+                                                    enrichedOrder.setProduct(product);
+                                                    enrichedOrder.setCustomer(customerOrder.customer);
+                                                    enrichedOrder.setOrder(customerOrder.order);
+                                                    return enrichedOrder;
+                                                })
+                        )
+                )
+        );
+    }
+----
+
+Here we have 3 inputs. The first function takes a `KStream` and its output is another `Function` that takes a `GlobalKTable` as its input and another function as its output.
```

Comment:
We could do a similar (visual) mapping like in the previous comment for clarity.